dashboard/app/jobs/evaluate_rubric_job.rb

```diff
@@ -86,6 +86,40 @@ class EvaluateRubricJob < ApplicationJob
     raise exception
   end
 
+  rescue_from(ProfanityFilterException) do |exception|
+    if rack_env?(:development)
+      puts "EvaluateRubricJob Filter Error: #{exception.full_message} Type: #{exception.share_failure.type}"
+    end
+
+    # Record the failure, if we can
+    begin
+      rubric_ai_evaluation = pass_in_or_create_rubric_ai_evaluation(self)
+      rubric_ai_evaluation.status = SharedConstants::RUBRIC_AI_EVALUATION_STATUS[:PROFANITY_VIOLATION]
+      rubric_ai_evaluation.save!
+    rescue StandardError
+      # Ignore cascading errors when the rubric record does not exist
```

Comment:
It would shadow the original exception... and if an error occurs before the rubric ai evaluation was made, then the reason the record wouldn't exist is because of the original error. my thought was that the originating error was the important one to log.

An RAE record doesn't exist when it cannot find the ScriptLevel or Rubric or User and if any channel id errors happen... because the rubric evaluation record requires the project_id and rubric_id. These things shouldn't happen in practice because the controller action that is scheduling the jobs should also have to look these up to authorize the action.

Reply:
Only the final exception makes it to honey badger

Reply:
hmm are you saying the original exception (ProfanityFilterException) will automatically be reraised by the rescue_from block? I would bet that this is not the case because it seems dissimilar to the concept of rescuing in ruby.

> These things shouldn't happen in practice because the controller action that is scheduling the jobs should also have to look these up to authorize the action.

If something should not happen in practice, then we should surface exception when it does happen rather than squashing it, since it means an error on our part in either programming or design.